File: 2021/12/12/total_money.py
"""
https://leetcode.com/problems/calculate-money-in-leetcode-bank/
Author: Andrew Jarombek
Date: 12/12/2021
"""


def total_money_faster(n: int) -> int:
    result = 0
    loop = 0

    while n > 7:
        result += 28 + 7 * loop
        n -= 7
        loop += 1

    for i in range(1, n + 1):
        result += i + loop

    return result


def total_money(n: int) -> int:
    result = 0
    prev = 6
    i = 0

    while i < n:
        if i % 7 == 0:
            prev -= 5
            result += prev
        else:
            prev += 1
            result += prev

        print(prev)
        i += 1

    return result

File: 2022/08/27/java/TotalMoney.java
/**
 * https://leetcode.com/problems/calculate-money-in-leetcode-bank/
 * @author Andrew Jarombek
 * @since 8/27/2022
 */

public class TotalMoney {
    public static int totalMoney(int n) {
        int result = 1;
        int deposit = 1;

        for (int i = 1; i < n; i++) {
            if (i % 7 == 0)
                deposit -= 6;

            deposit++;
            result += deposit;
        }

        return result;
    }

    public static void main(String[] args) {
        assert totalMoney(4) == 10;
        assert totalMoney(10) == 37;
        assert totalMoney(20) == 96;
    }
}
